src/circuit/gadget/ecc/chip.rs

```diff
@@ -365,23 +368,24 @@ pub struct EccScalarFixedShort {
 /// `base_field_elem`.
 #[derive(Clone, Debug)]
 struct EccBaseFieldElemFixed {
-    base_field_elem: CellValue<pallas::Base>,
-    running_sum: ArrayVec<CellValue<pallas::Base>, { constants::NUM_WINDOWS + 1 }>,
+    base_field_elem: AssignedCell<pallas::Base, pallas::Base>,
+    running_sum: ArrayVec<AssignedCell<pallas::Base, pallas::Base>, { constants::NUM_WINDOWS + 1 }>,
 }
 
 impl EccBaseFieldElemFixed {
-    fn base_field_elem(&self) -> CellValue<pallas::Base> {
-        self.base_field_elem
+    fn base_field_elem(&self) -> AssignedCell<pallas::Base, pallas::Base> {
+        self.base_field_elem.clone()
     }
 }
 
 impl EccInstructions<pallas::Affine> for EccChip {
     type ScalarFixed = EccScalarFixed;
     type ScalarFixedShort = EccScalarFixedShort;
-    type ScalarVar = CellValue<pallas::Base>;
+    // TODO: V = pallas::Scalar?
```

Comment:
No; ScalarVar is used for variable-base scalar multiplication where the scalar is represented as a base element in F<sub>p</sub>, and the implementation assumes that p < q. So `V = pallas::Base` is correct.

```suggestion
```